src/DataBase/GoalDatabaseSql.cs

```diff
@@ -148,56 +159,105 @@ public async Task<int> MoveGoalToExpensesAsync(IGoal goal, IEntry expense)
 
         public async Task<int> MoveGoalToExpensesAsync(int goalId, int userId, IEntry expense)
         {
-            var dbGoal = await Database.Goals.FirstOrDefaultAsync(x => x.Id == goalId && x.UserId == userId);
-            Database.Goals.Remove(dbGoal ?? throw new InvalidOperationException());
+            var sqlConnection = new SqlConnection(Database.Database.GetDbConnection().ConnectionString);
+
+            if (sqlConnection.State == ConnectionState.Closed)
+            {
+                await sqlConnection.OpenAsync();
+            }
+
+            var sqlCommand = new SqlCommand("DELETE FROM Goals WHERE Id = @Id AND UserId = @UserId", sqlConnection)
+            {
+                CommandType = CommandType.Text
+            };
+            sqlCommand.Parameters.AddWithValue("@Id", goalId);
+            sqlCommand.Parameters.AddWithValue("@UserId", userId);
+            await sqlCommand.ExecuteNonQueryAsync();
+
             var dbExpense = new ExpenseModel(expense, userId);
-            await Database.Expenses.AddAsync(dbExpense);
-            await Database.SaveChangesAsync();
+            sqlCommand =
+                new SqlCommand(
+                    "INSERT INTO Expenses(UserId, Amount, Title, Date, IsMonthly, Importance, Currency) VALUES (@UserId, @Amount, @Title, @Date, @IsMonthly, @Importance, @Currency);SELECT CAST(scope_identity() AS int);",
+                    sqlConnection)
+                { CommandType = CommandType.Text };
+            sqlCommand.Parameters.AddWithValue("@UserId", userId);
+            sqlCommand.Parameters.AddWithValue("@Amount", expense.Amount);
+            sqlCommand.Parameters.AddWithValue("@Title", expense.Title);
+            sqlCommand.Parameters.AddWithValue("@Date", expense.Date);
+            sqlCommand.Parameters.AddWithValue("@IsMonthly", expense.Recurring);
+            sqlCommand.Parameters.AddWithValue("@Importance", expense.Importance);
+            sqlCommand.Parameters.AddWithValue("@Currency", "EUR");
+            //public DateTime Date { get; set; }
+            //public bool IsMonthly { get; set; }
+            //public int Importance { get; set; }
```

Comment:
Is this commented code will be used in the future?

Reply:
Nope, just forgot to delete :D